### PR TITLE
CORE-8812 Retrieving relevant states

### PR DESCRIPTION
**Pull request by mbrkic-r3**
- Added Avro record for finding relevant states of ledger transaction

### CODE REVIEW

File: ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/UtxoLedgerService.kt [hunk 1]
```diff
@@ -77,6 +77,17 @@ interface UtxoLedgerService {
...
+    /**
+     * Finds unconsumed states of transaction in the vault, resolves it to a ledger transaction and returns it
+     *
+     * @param id The id of the transaction to find
+     * @param stateClass State type
+     *
+     * @return The unconsumed states (or empty list if not found).
+     */
+    @Suspendable
+    fun <T: ContractState> findUnconsumedStatesByType(id: SecureHash, stateClass: Class<out T>): List<StateAndRef<T>>
```
Comment (blsemo) [inline ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/UtxoLedgerService.kt]: Why are we looking by transaction id? I thought idea was to find any consumable states of a certain type, whithout knowing their transaction id?

Comment (mbrkic-r3) [reply ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/UtxoLedgerService.kt]: fixed

**Review by blsemo — DISMISSED** [review]
LGTM

**Review by blsemo — APPROVED** [review]
Re-approved

**Pull request merged**